base/loading.jl

```diff
@@ -415,14 +415,15 @@ end
 evalfile(path::AbstractString, args::Vector) = evalfile(path, UTF8String[args...])
 
 function create_expr_cache(input::AbstractString, output::AbstractString)
-    isfile(output) && rm(output)
```

Comment:
Okay. Looking at the code again: Maybe all that is necessary is to change this line to
```
try rm(output) end
```
?

If `.ji` files are already created atomically, then the only remaining issue is that `rm` may fail if run in parallel.